hub/api/tests/test_api.py

```diff
@@ -122,6 +112,77 @@ def test_iterate_dataset(ds):
         assert label == labels[idx]
 
 
+def test_compute_slices(memory_ds):
+    ds = memory_ds
```

Comment:
nit-pick can be simplified by using test-cases/data providers.
Also can you add some tests examples from Indexing explained here: https://www.tensorflow.org/guide/tensor#indexing

Reply:
@imshashank It's not worth it to use test cases here, simply because we want to test the syntax in addition to the implementation. Passing in tuples instead of the arguments one-by-one can cause some un-intended side-effects.

Reply:
That's fine with me.